client/gogeta_client.go

```diff
@@ -118,45 +130,108 @@ func (client *Gogeta) RunGogetaClient() *sourcesystem.SourceRepository {
 	client.queueMessages()
 
 	if len(client.data) <= 0 {
-		noQueueData := fmt.Sprintf("No queue message data received from %v", os.Getenv(config.QueueURL))
-		client.Log.Info(noQueueData)
+		client.Log.Info("No data received from queue to clone project")
 		return &repo
 	}
 
-	client.setVersionControl()
-	if client.SCM == nil {
-		noSCM := fmt.Sprintf("No SCM could be found for %v", client.data[0].RepoType)
-		client.Log.Info(noSCM)
+	client.broadcastProgress("Source code download request received")
+
+	if err := client.setVersionControl(); err != nil {
+		client.broadcastFailure(err.Error(), "client.SCM value is nil")
 		return &repo
 	}
-	downloadMessage := fmt.Sprintf("Getting %v source for build ID %v", client.data[0].RepoType, client.data[0].ID)
-	client.Log.Info(downloadMessage)
 
-	client.sendGamebuildrMessage(downloadMessage, 0)
+	client.broadcastProgress("Downloading latest project source")
+
 	if err := client.downloadSource(&repo); err != nil {
-		client.sendGamebuildrMessage(err.Error(), 1)
-		client.Log.Error(err.Error())
+		cloneErr := fmt.Sprintf("Cloning failed with the following error: %v", err.Error())
+		client.broadcastFailure(cloneErr, err.Error())
 		return &repo
```

Comment:
This is the only case where I think we should send the error back to the user. i.e. "Cloning failed: Repo Size too large", "Cloning Failed: Source URL could not be found" etc.